### ISSUE TITLE
LZO implementation is GPL-licensed

### ISSUE THREAD

**Issue by gniezen** [schroffl/node-lzo#12]
Hi there,

I noticed that node-lzo is licensed under the MIT license, but the LZO implementation used is GPL. Wouldn't that make node-lzo GPL-licensed as well?

**Comment by schroffl** [comment]
I guess you are right, I never really looked at the legal side of software development.
Do you have any advise other than updating the license? 

**Edit**: What's with the older versions of the library, do I have to update their license as well? It would be annoying to purge everything from the git repository.

**Comment by gniezen** [comment]
Unfortunately not. We have BSD-2 licensed code and it would've been great to use node-lzo. Having a GPL-licensed dependency shouldn't be an issue for us, as our code is already open-source, but but we want people to be able to use our code in closed-source apps as well.

I think we'll have to switch to an LZO implementation with a more permissive license (MIT/BSD) or LGPL. I have not been able to find any MIT/BSD licensed implementations, but there are LGPL implementations in for example ffmpeg.

**Comment by schroffl** [comment]
I am sorry for that. The only way you could use this is if I were to write the LZO implementation on my own or used a library that works with the MIT License, right?

**Comment by gniezen** [comment]
Yes, an MIT- or BSD-licensed library would be ideal, but the only one I've found is written in C#. You could also used a LGPL-licensed implentation. I've started working on building a module that makes use of the one in FFmpeg's avutil C library, using N-API to wrap the calls.

**Comment by schroffl** [comment]
I actually planned on writing a pure JavaScript implementation when I kicked off this library, but I was very much too inexperienced with C and not motivated enough to dig through the code and translate everything to JS. It also doesn't help that everything is written in a rather obfuscated manner.
Do you know of any resources that outline the process? Maybe even pseudocode? I wasn't able to find anything back then and I don't think it has gotten a lot better.

**Comment by gniezen** [comment]
I do indeed: https://www.kernel.org/doc/Documentation/lzo.txt It's a description of how the LZO decompression algorithm in the Linux kernel works.

**Comment by schroffl** [comment]
Oh cool, I will definitely take a look at it. Thank you!
